discovery/db_discovery.go

```diff
@@ -0,0 +1,100 @@
+package discovery
+
+import (
+	"time"
+
+	"github.com/livepeer/go-livepeer/common"
+	"github.com/livepeer/go-livepeer/core"
+	lpTypes "github.com/livepeer/go-livepeer/eth/types"
+	"github.com/livepeer/go-livepeer/net"
+
+	"github.com/golang/glog"
+)
+
+var cacheRefreshInterval = 1 * time.Hour
+var getTicker = func() *time.Ticker {
+	return time.NewTicker(cacheRefreshInterval)
+}
+
+type DBOrchestratorPoolCache struct {
+	node *core.LivepeerNode
+}
+
+func NewDBOrchestratorPoolCache(node *core.LivepeerNode) *DBOrchestratorPoolCache {
+	if node.Eth == nil {
+		glog.Error("Could not refresh DB list of orchestrators: LivepeerNode nil")
+		return nil
+	}
+
+	ticker := getTicker()
+	go func(node *core.LivepeerNode) {
+		for {
+			select {
+			case <-ticker.C:
+				orchestrators, err := node.Eth.RegisteredTranscoders()
+				if err != nil {
+					glog.Error("Could not refresh DB list of orchestrators: ", err)
+					continue
+				}
+				_, dbOrchErr := cacheDBOrchs(node, orchestrators)
+				if dbOrchErr != nil {
+					glog.Error("Could not refresh DB list of orchestrators: cacheDBOrchs err")
+					continue
+				}
+			}
+		}
+	}(node)
+
+	return &DBOrchestratorPoolCache{node: node}
+}
+
+func (dbo *DBOrchestratorPoolCache) GetOrchestrators(numOrchestrators int) ([]*net.OrchestratorInfo, error) {
+	orchs, err := dbo.node.Database.SelectOrchs()
+	if err != nil || len(orchs) <= 0 {
+		return nil, err
+	}
+
+	var uris []string
+	for _, orch := range orchs {
+		uri := orch.ServiceURI
+		uris = append(uris, uri)
+	}
+
+	orchPool := NewOrchestratorPool(dbo.node, uris)
+
+	orchInfo, err := orchPool.GetOrchestrators(numOrchestrators)
```

Comment:
Should `orchInfo` be plural is it is an array?

Reply:
Yes! I also changed `uri` in the `orchestratorPool` struct to `uris`, which makes more sense as well